src/features/tempRole/temporaryRoleCommand.ts

```diff
@@ -0,0 +1,91 @@
+import {
+  ApplicationCommandOptionType,
+  ApplicationCommandType,
+  GuildMember,
+  PermissionsBitField,
+  Role,
+  time,
+} from 'discord.js'
+
+import parse from 'parse-duration'
+
+import { prisma } from '@/prisma'
+import { defineCommand } from '@/types/defineCommand'
+
+export const temporaryRoleCommand = defineCommand({
+  data: {
+    name: 'temp-role',
+    description: 'Set a temporary role for a user',
+    defaultMemberPermissions: PermissionsBitField.Flags.ManageRoles,
+    type: ApplicationCommandType.ChatInput,
+    options: [
+      {
+        name: 'user',
+        description: 'User to set the temporary role for',
+        type: ApplicationCommandOptionType.User,
+        required: true,
+      },
+      {
+        name: 'role',
+        description: 'Role to set',
+        type: ApplicationCommandOptionType.Role,
+        required: true,
+      },
+      {
+        name: 'duration',
+        description: 'Duration of the temporary role e.g. 1d',
+        type: ApplicationCommandOptionType.String,
+        required: true,
+      },
+    ],
+  },
+  ephemeral: true,
+  execute: async (botContext, interaction) => {
+    if (!interaction.guild || !interaction.isChatInputCommand()) return
+
+    // Get user from the interaction
+    const user = interaction.options.getMember('user') as GuildMember
+
+    // Get role from the interaction
+    const role = interaction.options.getRole('role') as Role
+
+    // Get duration from the interaction
+    const duration = interaction.options.getString('duration') as string
+    const durationMs = parse(duration)
+
+    if (durationMs === null) {
```

Comment:
What if I input `0s` lol, `!durationMs` should be able to cover both `durationMs === null` and `durationMs === 0`